tools/rosclean/src/rosclean/__init__.py

```diff
@@ -138,20 +155,69 @@ def get_disk_usage(d):
     else:
         raise CleanupException("rosclean is not supported on this platform")
 
+def _sort_file_by_oldest(d):
+    """
+    Get files and directories in specified path sorted by last modified time
+    :param d: directory path, ```str```
+    :return:  a list of files and directories sorted by last modified time (old first), ```list```
+    """
+    files = os.listdir(d)
+    files.sort(key=lambda f: os.path.getmtime(os.path.join(d, f)))
+    return files
+
+def _check_delete_file_size(log_size, d, f_name):
+    """
+    Get size of specified directory after deleting specified file or directory
+    :param log_size: size of log directory in bytes, ```int```
+    :param d: directory path, ```str```
+    :param file: file or directory, ```str```
+    :return: size of directory after purging file, ```int```
+    """
+    target = os.path.join(d, f_name)
+    f_size = 0
+    if os.path.isdir(target):
+        for dir_path, dir_name, filenames in os.walk(target):
+            for f in filenames:
+                fp = os.path.join(dir_path, f)
+                f_size += os.path.getsize(fp)
+    else:
+        f_size = os.path.getsize(target)
+    return (log_size - f_size)
```

Comment:
This function doesn't actually care about `log_size`. I suggest removing the argument and simply return `f_size`. The calling code can then do the simple subtraction.

Isn't the function obsolete then since `get_disk_usage(target)` provides the same information?